### PR TITLE
feat(presets/regex-managers): improve preset for dockerfile

### PR DESCRIPTION
**Pull request by jhaumont**
## Changes

Change regexp for `versioning` capturing group.
Add a new capturing group for `registryUrl` (I hesitated for this one)

## Context

Improve of regex-managers preset for dockerfile:
* extend possibility for `versioning` setup (have the possibility to use a regexp like this: `regex:^(?<compatibility>\w+)-(?<minor>\d+)-(?<variant>\w+)$`)
* add the possibility to have a custom registry define in `registryUrl` capturing group

Examples in https://regex101.com/r/JxpUbY/1

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
why versioning is changed?

**Comment by jhaumont** [comment]
> why versioning is changed?

with the actual version of versioning regexp(`(?<versioning>[a-z-0-9]+?)`), I can't do something like `versioning=regex:^(?<compatibility>\w+)-(?<minor>\d+)-(?<variant>\w+)$` (I have put an example on https://regex101.com/r/JxpUbY/1)

**Comment by viceice** [comment]
> > why versioning is changed?
> 
> with the actual version of versioning regexp(`(?<versioning>[a-z-0-9]+?)`), I can't do something like `versioning=regex:^(?<compatibility>\w+)-(?<minor>\d+)-(?<variant>\w+)$` (I have put an example on https://regex101.com/r/JxpUbY/1)

OK, but `variant` isn't a supported regex versioning keyword, so the group should be removed 😉

File: lib/config/presets/internal/regex-managers.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export const presets: Record<string, Preset> = {
       {
         fileMatch: ['(^|/|\\.)Dockerfile$', '(^|/)Dockerfile[^/]*$'],
         matchStrings: [
-          '# renovate: datasource=(?<datasource>[a-z-]+?) depName=(?<depName>[^\\s]+?)(?: (lookupName|packageName)=(?<packageName>[^\\s]+?))?(?: versioning=(?<versioning>[a-z-0-9]+?))?\\s(?:ENV|ARG) .+?_VERSION[ =]"?(?<currentValue>.+?)"?\\s',
+          '# renovate: datasource=(?<datasource>[a-z-]+?)(?: registryUrl=(?<registryUrl>[^\\s]+?))? depName=(?<depName>[^\\s]+?)(?: (lookupName|packageName)=(?<packageName>[^\\s]+?))?(?: versioning=(?<versioning>[^\\s]+?))?\\s(?:ENV|ARG) .+?_VERSION[ =]"?(?<currentValue>.+?)"?\\s',
```
Comment (viceice) [inline lib/config/presets/internal/regex-managers.ts:12]: I would prefer to move `registryUrl` to the end, as it's rarly used
```suggestion
          '# renovate: datasource=(?<datasource>[a-z-]+?) depName=(?<depName>[^\\s]+?)(?: (lookupName|packageName)=(?<packageName>[^\\s]+?))?(?: versioning=(?<versioning>[^\\s]+?))?(?: registryUrl=(?<registryUrl>[^\\s]+?))?\\s(?:ENV|ARG) .+?_VERSION[ =]"?(?<currentValue>.+?)"?\\s',
```

**Comment by jhaumont** [comment]
> > > why versioning is changed?
> > 
> > 
> > with the actual version of versioning regexp(`(?<versioning>[a-z-0-9]+?)`), I can't do something like `versioning=regex:^(?<compatibility>\w+)-(?<minor>\d+)-(?<variant>\w+)$` (I have put an example on https://regex101.com/r/JxpUbY/1)
> 
> OK, but `variant` isn't a supported regex versioning keyword, so the group should be removed 😉

😮 I don't know that. Thanks !

File: lib/config/presets/internal/regex-managers.ts (hunk 1 above)
Comment (jhaumont) [reply lib/config/presets/internal/regex-managers.ts]: done

**Review by viceice — APPROVED** [review]

**Comment by jhaumont** [comment]
@viceice thx for the approval! you handle the merge?

**Pull request merged**